Makefile

```diff
@@ -1,137 +1,205 @@
-PHP_CS_FIXER_FUTURE_MODE=1
-PHPSTAN=./phpstan.phar
-PHP-CS-FIXER=./php-cs-fixer-v2.phar
-PHPUNIT=vendor/bin/phpunit
-INFECTION=build/bin/infection.phar
+.DEFAULT_GOAL := help
+
+.PHONY: help
+# TODO: add a check to ensure the result is always the expected one
+# TODO: add a check to ensure whenever a PHONY is declared, the following target has the same name; e.g. here .PHONY=help
+# so the next target should be `help:`. Otherwise this is most certainly a mistake.
+help:
+	@echo "\033[33mUsage:\033[0m\n  make TARGET\n\n\033[32m#\n# Commands\n#---------------------------------------------------------------------------\033[0m\n"
+	@fgrep -h "##" $(MAKEFILE_LIST) | fgrep -v fgrep | sed -e 's/\\$$//' | sed -e 's/##//' | awk 'BEGIN {FS = ":"}; {printf "\033[33m%s:\033[0m%s\n", $$1, $$2}'
+
 
-# URLs to download all tools
-BOX_URL="https://github.com/humbug/box/releases/download/3.1.0/box.phar"
-PHP-CS-FIXER_URL="https://cs.sensiolabs.org/download/php-cs-fixer-v2.phar"
+#
+# Variables
+#---------------------------------------------------------------------------
+BOX=./.tools/box
+BOX_URL="https://github.com/humbug/box/releases/download/3.8.0/box.phar"
+
+PHP_CS_FIXER=./.tools/php-cs-fixer
+PHP_CS_FIXER_URL="https://cs.sensiolabs.org/download/php-cs-fixer-v2.phar"
+PHP_CS_FIXER_CACHE=build/cache/.php_cs.cache
+
+PHPSTAN=./.tools/phpstan
 PHPSTAN_URL="https://github.com/phpstan/phpstan/releases/download/0.11.5/phpstan.phar"
 
-FLOCK=./devTools/flock
+PHPUNIT=vendor/bin/phpunit
+
+INFECTION=./build/infection.phar
 
-DOCKER_RUN=docker run -t --rm -v "$$PWD":/opt/infection -w /opt/infection
+DOCKER_RUN=docker run --tty --rm --volume "$$PWD":/opt/infection --workdir /opt/infection
 DOCKER_RUN_72=$(FLOCK) devTools/*php72*.json $(DOCKER_RUN) infection_php72
+DOCKER_RUN_72_IMAGE=devTools/Dockerfile-php72-xdebug.json
 DOCKER_RUN_73=$(FLOCK) devTools/*php73*.json $(DOCKER_RUN) infection_php73
+DOCKER_RUN_73_IMAGE=devTools/Dockerfile-php73-xdebug.json
 
-.PHONY: all
-#Run all checks, default when running 'make'
-all: analyze test
+FLOCK=./devTools/flock
 
-#Non phony targets for phars etc.
-vendor: composer.json composer.lock
-	composer install
 
-build/cache:
-	mkdir -p build/cache
+#
+# Commands (phony targets)
+#---------------------------------------------------------------------------
 
-./php-cs-fixer-v2.phar:
-	wget $(PHP-CS-FIXER_URL)
-	chmod a+x ./php-cs-fixer-v2.phar
+.PHONY: compile
+compile:	## Bundles Infection into a PHAR
+compile: $(INFECTION)
 
-./phpstan.phar:
-	wget $(PHPSTAN_URL)
-	chmod a+x ./phpstan.phar
+.PHONY: cs-fix
+cs-fix:	  	## Runs PHP-CS-Fixer
+cs-fix: $(PHP_CS_FIXER)
```

Comment:
I quite didn't get when you are expected to run `cs-fix`. I get the use of `cs-check` for the CI but is there a case where you prefer `cs-check` over `cs-fix` locally?

Reply:
I think this could be simplified to use `cs` and fix the CS when expected: there is already a dedicated CS check on a PR basis I don't think it provides anything to check it locally as well...

thoughts @infection/core?

Reply:
I'd love to use `cs` instead, because it just happens that's how this rule usually named elsewhere.